var/spack/repos/builtin/packages/py-tensorflow/package.py

```diff
@@ -0,0 +1,692 @@
+            # only supports compute capabilities >= 3.5
+            if spec.variants['cuda_arch'].value != 'none':
+                capabilities = ','.join('{0:.1f}'.format(
+                    float(i) / 10.0) for i in spec.variants['cuda_arch'].value)
+                env.set('TF_CUDA_COMPUTE_CAPABILITIES', capabilities)
+        else:
+            env.set('TF_NEED_CUDA', '0')
+
+        # Do you wish to download a fresh release of clang? (Experimental)
+        env.set('TF_DOWNLOAD_CLANG', '0')
+
+        # Do you wish to build TensorFlow with MPI support?
+        if '+mpi' in spec:
+            env.set('TF_NEED_MPI', '1')
+
+            # Please specify the MPI toolkit folder
+            env.set('MPI_HOME', spec['mpi'].prefix)
+        else:
+            env.set('TF_NEED_MPI', '0')
+            env.unset('MPI_HOME')
+
+        # Please specify optimization flags to use during compilation when
+        # bazel option '--config=opt' is specified
+        env.set('CC_OPT_FLAGS', spec.target.optimization_flags(
+            spec.compiler.name, spec.compiler.version))
+
+        # Would you like to interactively configure ./WORKSPACE for
+        # Android builds?
+        if '+android' in spec:
+            env.set('TF_SET_ANDROID_WORKSPACE', '1')
+
+            # Please specify the home path of the Android NDK to use
+            env.set('ANDROID_NDK_HOME', spec['android-ndk'].prefix)
+            env.set('ANDROID_NDK_API_LEVEL', spec['android-ndk'].version)
+
+            # Please specify the home path of the Android SDK to use
+            env.set('ANDROID_SDK_HOME', spec['android-sdk'].prefix)
+            env.set('ANDROID_SDK_API_LEVEL', spec['android-sdk'].version)
+
+            # Please specify the Android SDK API level to use
+            env.set('ANDROID_API_LEVEL', spec['android-sdk'].version)
+
+            # Please specify an Android build tools version to use
+            env.set('ANDROID_BUILD_TOOLS_VERSION', spec['android-sdk'].version)
+        else:
+            env.set('TF_SET_ANDROID_WORKSPACE', '0')
+
+        # Do you wish to build TensorFlow with iOS support?
+        if '+ios' in spec:
+            env.set('TF_CONFIGURE_IOS', '1')
+        else:
+            env.set('TF_CONFIGURE_IOS', '0')
+
+        # set tmpdir to a non-NFS filesystem
+        # (because bazel uses ~/.cache/bazel)
+        # TODO: This should be checked for non-nfsy filesystem, but the current
+        #       best idea for it is to check
+        #           subprocess.call([
+        #               'stat', '--file-system', '--format=%T', tmp_path
+        #       ])
+        #       to not be nfs. This is only valid for Linux and we'd like to
+        #       stay at least also OSX compatible
+        tmp_path = '/tmp/spack/tf'
+        mkdirp(tmp_path)
+        env.set('TEST_TMPDIR', tmp_path)
+        # TODO: Is setting this necessary? It breaks `spack build-env`
+        # because Bash can't find my .bashrc
+        env.set('HOME', tmp_path)
```

Comment:
It looks like `bazel` has a `--output_user_root` option you can use to change where things are written instead of `$HOME`.

Reply:
See what I did to the `bazel` package for an example of its usage.